src/pages/library/DataExplorer.js

```diff
@@ -37,21 +114,58 @@ const DataExplorer = props => {
             // Don't call Nav.history.replace(). That will trigger a request and
             // cause the page to flicker.
             const url = window.location.origin + '#' + Nav.history.location.pathname.slice(1) + '?' + message.deQueryStr
-            window.history.replaceState({}, 'Data Explorer - ' + props.dataset, url)
+            window.history.replaceState({}, 'Data Explorer - ' + dataset, url)
           }
         }
       }
     })
-  ])
-}
 
+    const paragraphStyle = { style: { margin: '2rem' } }
+    const notInAuthDomainError = div({
+      style: {
+        fontSize: 18,
+        margin: '3rem',
+        width: 800
+      }
+    }, [
+      div(paragraphStyle, ['This Data Explorer requires you to be in the ',
+        span({ style: { fontWeight: 'bold' } }, this.authDomain),
+        ' Terra group.']),
+      div(paragraphStyle, 'If you have a different Google account in that ' +
+        'group, please sign out of Terra and sign in with that account. To ' +
+        'sign out of Terra, click on the menu on the upper left, ' +
+        'click on your name, then click Sign Out'),
+      div(paragraphStyle, [
+        'If you don\'t have a Google account in that group, please ',
+        link({ onClick: () => { contactUsActive.set(true) } }, 'apply for access.')
+      ])
+    ])
+
+    return h(Fragment, [
+      header,
+      Utils.cond(
+        [this.authDomain === undefined, dataExplorer],
+        [groups === undefined || completedDeOauth === undefined, centeredSpinner],
+        [groups && groups.includes(this.authDomain) && completedDeOauth === false, () => { window.open(this.origin, '_self') }],
```

Comment:
It's not a good idea to perform mutations in the render path. Even though this effectively tears down the page, we should still do this in e.g. `componentDidUpdate` for consistency.